### PR TITLE
[#1749] Admin group membership endpoints (add / remove / role-change)

### LINKED ISSUE #1749: [admin/be] Group membership admin endpoints (add / remove / role-change)
## Parent

Part of #1744 (system-wide admin section). Depends on admin BE foundation and admin BE groups endpoints.

## Goal

Admin can add, remove, and change role of a group's members, bypassing the per-group `requireGroupAdmin` middleware but still respecting the existing invariants.

### Endpoints

- `POST /api/v1/admin/groups/{groupID}/members` — body `{ "userID": "<uuid>", "role": "viewer|user|admin|owner" }`. Creates membership via `GroupMembershipRegistry.CreateUnderCap(...)`. If the cap is hit, return the existing `ErrTooManyGroupMemberships`.
- `DELETE /api/v1/admin/groups/{groupID}/members/{userID}` — wraps `DeleteWithMemberInvariants`. Returns `ErrLastOwner` / `ErrLastAdmin` / `ErrLastMember` unchanged.
- `PATCH /api/v1/admin/groups/{groupID}/members/{userID}` — body `{ "role": "..." }`; wraps `UpdateRoleWithMemberInvariants`.

### Notes

- Admin endpoints reuse the same service methods to keep invariants single-sourced.
- A member added by an admin gets a synthetic `joined_at = now()`. No invite-token flow — admin add is direct.
- Cross-tenant safety: refuse to add a user whose `tenant_id` differs from the group's tenant. Return a typed `ErrTenantMismatch`.

## Acceptance criteria

- [ ] Three endpoints + Swagger + FE codegen.
- [ ] Handler tests cover each invariant violation surfaces the correct sentinel.
- [ ] Cross-tenant add rejected with `ErrTenantMismatch`.
- [ ] Audit-log entries for add / remove / role-change.

## References

- Umbrella: #1744
- `GroupService` invariant guards (`ErrLastAdmin` / `ErrLastOwner` / `ErrLastMember`)
- `registry.GroupMembershipRegistry`

### PR DESCRIPTION
**Pull request by denisvmedia**
## Summary

Implements #1749 — three system-admin endpoints under `/api/v1/admin/*` for managing a group's members, bypassing the per-group `requireGroupAdmin` middleware while reusing the same `GroupService`/registry methods so the invariants stay single-sourced.

| Method | Route | Behaviour |
|--------|-------|-----------|
| `POST` | `/api/v1/admin/groups/{groupID}/members` | Direct add via `GroupMembershipRegistry.CreateUnderCap` |
| `DELETE` | `/api/v1/admin/groups/{groupID}/members/{userID}` | Wraps `DeleteWithMemberInvariants` |
| `PATCH` | `/api/v1/admin/groups/{groupID}/members/{userID}` | Wraps `UpdateRoleWithMemberInvariants` |

## Details

- **New `GroupService.AdminAddMember`** routes the write through `createMembershipUnderCap`, so the per-user cap (`ErrTooManyGroupMemberships`) is enforced by the same atomic path as `AddMember`/`CreateGroup` — no parallel invariant code.
- Admin-added members get a synthetic `joined_at = now()` — no invite-token flow.
- **Cross-tenant safety:** adding a user whose `tenant_id` differs from the group's tenant is rejected with the new typed `ErrTenantMismatch` sentinel, mapped to `422 admin.member.tenant_mismatch`.
- Remove / role-change delegate to `GroupService.RemoveMember` / `UpdateMemberRole`, surfacing `ErrLastOwner` / `ErrLastMember` unchanged.
- Every action audit-logs via `AuditService` with admin user ID + group ID + tenant ID, matching the #1746/#1747 pattern.
- Swagger annotations + regenerated FE API types (`frontend/src/types/api.d.ts`) included.

## Acceptance criteria

- [x] Three endpoints + Swagger + FE codegen.
- [x] Handler tests cover each invariant violation surfacing the correct sentinel (`ErrLastOwner`, `ErrLastMember`, `ErrTooManyGroupMemberships`, `ErrAlreadyMember`).
- [x] Cross-tenant add rejected with `ErrTenantMismatch`.
- [x] Audit-log entries for add / remove / role-change (with tenant ID).

## Testing

- `go build ./...` — pass
- `golangci-lint run ./apiserver/... ./services/...` — 0 issues
- `go test ./apiserver/... ./services/... ./registry/...` — pass
- Handler tests cover happy paths, each invariant, cross-tenant rejection, non-admin 403, 404s, invalid role, and audit-row contents.

No migration required — no model changed.

Closes #1749

## Summary by CodeRabbit

* **New Features**
  * Admins can add members to groups, remove members, and change member roles via new admin group-membership endpoints.

* **Tests**
  * Added comprehensive tests covering add/remove/role-change flows, validation, audit logging, and edge-case invariants (e.g., last-owner/member protections).

* **Documentation**
  * API docs and schema definitions updated to document the new admin membership endpoints, request formats, responses, and error cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/denisvmedia/inventario/pull/1770?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: go/apiserver/admin_group_members.go [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+//
+// `userID` is the user to admit; `role` is the role they are admitted
+// with. Both are required — there is no invite-token flow here, an
+// admin add is direct, so the role cannot be deferred to an acceptance
+// step.
+type AdminAddMemberRequest struct {
+	// UserID is the ID of the user being added to the group.
+	UserID string `json:"userID"`
+	// Role is the group role granted to the user: viewer|user|admin|owner.
+	Role models.GroupRole `json:"role"`
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/admin_group_members.go]: Fixed in 1cc83f0b. Added `validate:"required"` to `AdminAddMemberRequest.UserID`/`Role` and `AdminUpdateMemberRoleRequest.Role`, matching the `AdminBlockRequest` / `FeedbackRequest` convention (swaggo consumes the `validate` tag in this repo). Regenerated swagger + FE codegen — `AdminAddMemberRequest` now has `required: [role, userID]` and `AdminUpdateMemberRoleRequest` has `required: [role]`, so the generated TS fields are non-optional.

File: frontend/src/types/api.d.ts [hunk 2]
```diff
@@ -7425,6 +7674,12 @@ export type paths = {
 export type webhooks = Record<string, never>;
 export type components = {
     schemas: {
+        "apiserver.AdminAddMemberRequest": {
+            /** @description Role is the group role granted to the user: viewer|user|admin|owner. */
+            role?: components["schemas"]["models.GroupRole"];
+            /** @description UserID is the ID of the user being added to the group. */
+            userID?: string;
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/types/api.d.ts]: Fixed in 1cc83f0b. Added `validate:"required"` to the underlying Go DTOs (`AdminAddMemberRequest.UserID`/`Role`, `AdminUpdateMemberRoleRequest.Role`) and regenerated. `frontend/src/types/api.d.ts` now emits `userID: string` and `role: components["schemas"]["models.GroupRole"]` (no trailing `?`) for both request types.

File: go/apiserver/admin_routes.go [hunk 3]
```diff
@@ -73,6 +81,11 @@ func Admin(params AdminParams) func(r chi.Router) {
 		blacklist:    params.Blacklist,
 		auditService: params.AuditService,
 	}
+	groupMembersAPI := &adminGroupMembersAPI{
+		factorySet:   params.FactorySet,
+		groupService: params.GroupService,
+		auditService: params.AuditService,
+	}
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/apiserver/admin_routes.go:96]: Fixed in 1cc83f0b. Added a constructor-time guard at the top of `Admin(...)`: `if params.GroupService == nil { panic("apiserver.Admin requires non-nil AdminParams.GroupService") }`. Note the existing constructor does not guard the other required deps (`FactorySet`, `Blacklist`, `AuditService`) — there was no established guard pattern to match, so this is a single fail-fast panic for the newly-added dependency, which is sufficient to surface the misconfiguration at startup.

File: go/docs/swagger.json [hunk 4]
```diff
@@ -6691,6 +6895,23 @@
...
+                            "$ref": "#/definitions/models.GroupRole"
+                        }
+                    ]
+                },
+                "userID": {
+                    "description": "UserID is the ID of the user being added to the group.",
+                    "type": "string"
+                }
+            }
+        },
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/docs/swagger.json:6918]: Fixed in 1cc83f0b. `swagger.json` is regenerated from the Go DTOs, which now carry `validate:"required"` tags — `apiserver.AdminAddMemberRequest` has `required: ["role", "userID"]` and `apiserver.AdminUpdateMemberRoleRequest` has `required: ["role"]`.

File: go/docs/swagger.yaml [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+  apiserver.AdminAddMemberRequest:
+    properties:
+      role:
+        allOf:
+        - $ref: '#/definitions/models.GroupRole'
+        description: 'Role is the group role granted to the user: viewer|user|admin|owner.'
+      userID:
+        description: UserID is the ID of the user being added to the group.
+        type: string
+    type: object
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/docs/swagger.yaml:15]: Fixed in 1cc83f0b. Both points addressed: the request schemas now declare `required` (`AdminAddMemberRequest` → `role, userID`; `AdminUpdateMemberRoleRequest` → `role`) via `validate:"required"` tags on the Go DTOs, and `AdminMemberView.role` is now typed as `models.GroupRole` so swagger emits a `$ref` to the `models.GroupRole` enum instead of a bare `string`. `models.GroupRole` is a string-underlying type, so the JSON wire format is unchanged. swagger.yaml/json and `api.d.ts` regenerated.
